internal/worker/controlleragentconfig/worker.go

```diff
@@ -148,15 +187,21 @@ func (w *configWorker) loop() error {
 		select {
 		case <-w.catacomb.Dying():
 			return w.catacomb.Err()
-		case <-ch:
+		case <-w.reloadRequested:
 			w.reportInternalState(stateReload)
 
-			w.cfg.Logger.Infof("SIGHUP received, reloading config")
+			// Empty channel of extra reload requests.
+			for len(w.reloadRequested) > 0 {
+				<-w.reloadRequested
+			}
```

Comment:
I think this is the wrong place to do this. Let's just bubble up the changes i.e. remove this.

Reply:
Removed.